### PR TITLE
New: Activity feed for sidebar

### PR DESCRIPTION
**Pull request by JustinHoldstock**
Ported to Flow from Box React UI.

TODO: Tests to Jest

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

**Comment by tonyjin** [comment]
Nice PR :)

File: package.json [hunk 1]
```diff
@@ -193,18 +199,26 @@
...
     "react-tether": "^0.6.1",
     "react-textarea-autosize": "^6.0.0",
     "react-virtualized": "^9.13.0",
-    "regenerator-runtime": "^0.11.1"
+    "regenerator-runtime": "^0.11.1",
+    "scroll-into-view-if-needed": "^1.5.0",
+    "tabbable": "^1.1.2"
+  },
+  "dependencies": {
+    "draft-js": "^0.10.5"
```
Comment (priyajeet) [inline package.json]: Move to dev/peer

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    translations,
+    inputState,
+    handlers
+}: Props): ReactNode => (
+    <ul className='box-ui-activity-feed-active-state'>
+        {items.map((item) => {
+            switch (item.type) {
+                case 'comment':
+                    return (
+                        <li className='box-ui-activity-feed-comment' key={item.type + item.id}>
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: Search/replace classname prefixes, from box-ui-* --> bcs-*

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            margin-bottom: 10px;
+        }
+    }
+
+    .box-ui-activity-feed-empty-state {
+        align-items: center;
+        display: flex;
+        flex-direction: column;
+
+        .empty-state-cta {
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss]: All classes should be pre-fixed, otherwise there will be quite a few conflicts with webapp, since they will also have old activity feed classes

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    .box-ui-activity-feed-comment-input {
+        background-color: $white;
+        margin-bottom: 4px;
+        max-height: 500px;
+        min-height: 93px;
+
+        &.is-disabled {
+            opacity: .4;
+            pointer-events: none;
+        }
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss]: same as above

File: src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+const EmptyState = ({ isLoading, showCommentMessage }: Props): ReactNode => (
+    <div className='box-ui-activity-feed-empty-state'>
+        <IconActivityFeedEmptyState />
+        {isLoading ? (
+            <LoadingIndicator />
+        ) : (
+            <div className='empty-state-cta'>
+                <b>
+                    <FormattedMessage {...messages.noActivity} />
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js]: `<FormattedMessage tagName='b' {...messages.noActivity} />` instead of wrapping `<b>`

File: src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                </b>
+                {showCommentMessage ? (
+                    <aside>
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js]: same, can be simplified to 

```showCommentMessage && <FormattedMessage tagName='aside' {...messages.noActivityCommentPrompt} />```

File: src/components/ContentSidebar/ActivityFeed/activity-feed/index.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export default from './ActivityFeed';
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/index.js]: ```export { default } from ...;```
and
```export { default as Foo } from ...;```

is the current correct syntax

File: src/components/ContentSidebar/ActivityFeed/icons/IconActivityFeedEmptyState.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+import AccessibleSVG from 'box-react-ui/lib/icons/accessible-svg';
+
+type Props = {
+    className: string,
+    width: number,
+    height: number
+};
+
+const IconActivityFeedEmptyState = ({ className = '', width = 90, height = 90 }: Props): ReactNode => (
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/icons/IconActivityFeedEmptyState.js]: All icons should come from box-react-ui, so this file should be moved there unless its already there.

File: src/components/ContentSidebar/ActivityFeed/keywords/index.js [hunk 9]
```diff
@@ -0,0 +1 @@
+export default from './Keywords';
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/keywords/index.js]: Same as above, as well for other files

File: src/components/ContentSidebar/ActivityFeed/messages.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    },
+    approvalAssignees: {
+        defaultMessage: 'Assignees',
+        description: 'Title for assignees input',
+        id: 'boxui.activityFeed.approvalAssignees'
+    },
+    approvalDueDate: {
+        defaultMessage: 'Due Date',
+        description: 'Title for approvers due date input',
+        id: 'boxui.activityFeed.approvalDueDate'
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/messages.js]: search/replace boxui ---> be in id prefix.
The key for the messages should also same as id. See the global messages.js.

We may have to move this file and merge it with the global messages.js, since there are build tools around it for publishing.

File: src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js (hunk 6 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/index.js (hunk 7 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/index.js]: Done. Out of curiosity, what's the reasoning behind this style?

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss (hunk 3 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss]: That was a doozy but, but I did not add for ALL classes. Only those that were A) referenced in a JS file as a class name and B) Had a corresponding entry in a SASS file. 

There are certain instances (ie. `.flyout-wrapper`, `.overlay-wrapper` etc.) that do not have a corresponding assignment in JS. I have left those as is as to not cause conflict. Let me know if I should remove those.

**Comment by JustinHoldstock** [comment]
@priyajeet, since out of date messages are hidden. Copying and pasting from "add bcs- prefix" comments.

"That was a doozy but, but I did not add for ALL classes. Only those that were A) referenced in a JS file as a class name and B) Had a corresponding entry in a SASS file.

There are certain instances (ie. .flyout-wrapper, .overlay-wrapper etc.) that do not have a corresponding assignment in JS. I have left those as is as to not cause conflict. Let me know if I should remove those."

**Comment by wenboyu2** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #254

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ActivityFeed/comment/__tests__/InlineEdit-test.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+
+import { InlineEditBase as InlineEdit } from '../InlineEdit';
+
+describe('features/activity-feed/comment/InlineEdit', () => {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/comment/__tests__/InlineEdit-test.js]: Rename to match test locations

File: src/components/ContentSidebar/ActivityFeed/task/Task.js [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+        comment_delete: boolean,
+        comment_edit: boolean
+    },
+    translatedTaggedMessage: string,
+    translations: Translations,
+    taggedMessage: string
+};
+
+// eslint-disable-next-line
+class Task extends Component<Props> {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/task/Task.js]: Can these extend PureComponent instead?

File: src/components/ContentSidebar/ActivityFeed/version/README.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+## Comment
+
+Comment primitive
+
+### Usage
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/version/README.md]: is this file needed?

File: src/components/ContentSidebar/ActivityFeed/utils/formatTaggedMessage.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+import type { Node } from 'react';
+import { Link } from 'box-react-ui/lib/components/link';
+import Mention from '../comment/Mention';
+
+// this regex matches one of the following regular expressions:
+// mentions: ([@＠﹫]\[[0-9]+:[^\]]+])
+// urls: (?:\b)((?:(?:ht|f)tps?:\/\/)[\w\._\-]+(:\d+)?(\/[\w\-_\.~\+\/#\?&%=:\[\]@!$'\(\)\*;,]*)?)
+// NOTE: There are useless escapes in the regex below, should probably remove them when safe
+// eslint-disable-next-line
+const splitRegex = /((?:[@＠﹫]\[[0-9]+:[^\]]+])|(?:\b(?:(?:ht|f)tps?:\/\/)[\w\._\-]+(?::\d+)?(?:\/[\w\-_\.~\+\/#\?&%=:\[\]@!$'\(\)\*;,]*)?))/gim;
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/utils/formatTaggedMessage.js]: 😮

File: src/components/ContentSidebar/ActivityFeed/task/README.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
+## Task
+
+Task primitive
+
+### Usage
+```js
+    import { Task } from 'box-react-ui';
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ActivityFeed/task/README.md]: is this file needed?

File: src/components/ContentSidebar/ActivityFeed/task/Task.js (hunk 12 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/task/Task.js]: I'll make a tech foundation ticket to do that

File: src/components/ContentSidebar/ActivityFeed/version/README.md (hunk 13 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/README.md]: Gar, forgot to remove READMEs. Doing it now

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss [hunk 16]
```diff
@@ -0,0 +1,90 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.scss]: This should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.scss [hunk 17]
```diff
@@ -0,0 +1,137 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.scss]: This should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/comment/Comment.scss [hunk 18]
```diff
@@ -0,0 +1,148 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/comment/Comment.scss]: This should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/keywords/Keywords.scss [hunk 19]
```diff
@@ -0,0 +1,52 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/keywords/Keywords.scss]: Should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/task/Task.scss [hunk 20]
```diff
@@ -0,0 +1,115 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/task/Task.scss]: Should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/version/Version.scss [hunk 21]
```diff
@@ -0,0 +1,39 @@
+@import '../../../styles/variables';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/version/Version.scss]: Should be `@import '../../../variables';`

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js [hunk 22]
```diff
@@ -0,0 +1,261 @@
...
+import type { Node } from 'react';
+import cx from 'classnames';
+import { EditorState } from 'draft-js';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+
+import Avatar from 'box-react-ui/lib/components/avatar/Avatar';
+import Form from 'box-react-ui/lib/components/form-elements/form/Form';
+import DraftJSMentionSelector, {
+    DraftMentionDecorator
+} from 'box-react-ui/lib/components/form-elements/draft-js-mention-selector/DraftJSMentionSelector';
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js]: from 'box-react-ui/lib/components/form-elements/draft-js-mention-selector';

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js [hunk 23]
```diff
@@ -0,0 +1,261 @@
...
+
+type Props = {
+    approverSelectorContacts: SelectorItems,
+    className: string,
+    createComment: Function,
+    createTask: Function,
+    updateTask: Function,
+    getApproverContactsWithQuery: Function,
+    getMentionContactsWithQuery: Function,
+    intl: intlShape.isRequired,
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js]: ^ thats not a flowtype, its the old proptype
I think use any, or see the flow-typed for react-intl

File: src/components/ContentSidebar/ActivityFeed/comment/InlineDelete.js [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+import PlainButton from 'box-react-ui/lib/components/plain-button';
+import { Flyout, Overlay } from 'box-react-ui/lib/components/flyout';
+import IconTrash from 'box-react-ui/lib/icons/general/IconTrash';
+
+import messages from '../../../messages';
+
+type Props = {
+    onDelete: Function,
+    id: string,
+    intl: intlShape.isRequired,
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/comment/InlineDelete.js]: same

File: src/components/ContentSidebar/ActivityFeed/version/CollapsedVersion.js [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+                )
+            }}
+        />
+    );
+}
+
+type Props = {
+    action: 'upload',
+    collaborators: { [collaborator_id: string]: User },
+    intl: intlShape.isRequired,
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/version/CollapsedVersion.js]: same

File: src/components/ContentSidebar/ActivityFeed/version/Version.js [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+        default:
+            return null;
+    }
+}
+
+type Props = {
+    action: 'delete' | 'restore' | 'upload',
+    createdBy: User,
+    id: string,
+    intl: intlShape.isRequired,
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/version/Version.js]: same

File: src/components/ContentSidebar/ActivityFeed/comment/InlineEdit.js [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+
+import PlainButton from 'box-react-ui/lib/components/plain-button';
+import IconPencil from 'box-react-ui/lib/icons/general/IconPencil';
+
+import messages from '../../../messages';
+
+type Props = {
+    toEdit: Function,
+    id: Function,
+    intl: intlShape.isRequired
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/comment/InlineEdit.js]: same

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 28]
```diff
@@ -0,0 +1,110 @@
...
+                        </li>
+                    );
+                case 'file_version_error':
+                    // we currently only display this if errorCode is tooManyVersions
+                    if (item.errorCode !== 'tooManyVersions') {
+                        return null;
+                    }
+                    return (
+                        <li className='bcs-version-item' key={item.type + item.errorCode}>
+                            {<VersionError {...item} />}
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 29]
```diff
@@ -0,0 +1,110 @@
...
+                        </li>
+                    );
+                case 'keywords':
+                    return (
+                        <li className='bcs-keywords-item' key={item.type + item.id}>
+                            {<Keywords {...item} />}
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 30]
```diff
@@ -0,0 +1,110 @@
...
+                                {...item}
+                                translations={translations}
+                                inputState={inputState}
+                                handlers={handlers}
+                            />
+                        </li>
+                    );
+                case 'task':
+                    return (
+                        <li className='bcs-activity-feed-task' key={item.type + item.id}>
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 31]
```diff
@@ -0,0 +1,188 @@
+/**
+ * @flow
+ * @file Component for Activity feed
+ */
+
+import React, { Component } from 'react';
+import type { Node } from 'react';
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 32]
```diff
@@ -0,0 +1,188 @@
...
+
+import ActiveState from './ActiveState';
+import ApprovalCommentForm from '../approval-comment-form';
+import EmptyState from './EmptyState';
+import type { User, SelectorItems } from '../../../../flowTypes';
+import type { Comments, Tasks, Contacts, Versions, Item, Translations } from '../activityFeedFlowTypes';
+
+import './ActivityFeed.scss';
+
+function collapseFeedState(feedState: Array<Item>): Array<Item> {
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js [hunk 33]
```diff
@@ -0,0 +1,34 @@
+/**
+ * @flow
+ * @file Component for Activity feed empty state
+ */
+
+import React from 'react';
+import type { Node } from 'react';
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/EmptyState.js]: done

File: src/components/messages.js [hunk 34]
```diff
@@ -524,6 +525,192 @@ const messages: { [string]: MessageDescriptor } = defineMessages({
         description: 'Default error message shown when upload fails',
         defaultMessage: 'Something went wrong with the upload. Please try again.'
     },
+    approvalAddAssignee: {
+        defaultMessage: 'Add an assignee',
+        description: 'Placeholder for approvers input',
+        id: 'be.activityFeed.approvalAddAssignee'
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/messages.js]: done

File: src/components/ContentSidebar/ActivityFeed/version/VersionError.js [hunk 35]
```diff
@@ -0,0 +1,35 @@
...
+    errorCode: string
+};
+
+const VersionError = ({ errorCode }: Props): Node => (
+    <div className='bcs-version error'>
+        <span className='bcs-version-message'>{getErrorMessage(errorCode)}</span>
+    </div>
+);
+
+VersionError.displayName = 'VersionError';
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/VersionError.js]: removed all displayNames

File: src/components/ContentSidebar/ActivityFeed/version/Version.js [hunk 36]
```diff
@@ -0,0 +1,87 @@
...
+                    type='button'
+                >
+                    <IconInfoInverted height={16} width={16} />
+                </PlainButton>
+            </span>
+        ) : null}
+    </div>
+);
+
+Version.displayName = 'Version';
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/Version.js]: done

File: src/components/ContentSidebar/ActivityFeed/task/Task.js [hunk 37]
```diff
@@ -0,0 +1,143 @@
...
+            onDelete,
+            onEdit,
+            onTaskAssignmentUpdate,
+            permissions,
+            taggedMessage,
+            translatedTaggedMessage,
+            translations
+        } = this.props;
+        return (
+            <div className={`bcs-task ${isPending || error ? 'bcs-is-pending' : ''}`}>
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/task/Task.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 38]
```diff
@@ -0,0 +1,110 @@
+/**
+ * @flow
+ * @file Active state component for Activity Feed
+ */
+import React from 'react';
+import type { Node } from 'react';
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: done

File: src/components/ContentSidebar/ActivityFeed/version/VersionError.js [hunk 39]
```diff
@@ -0,0 +1,35 @@
...
+import type { Node } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import messages from '../../../messages';
+
+import './Version.scss';
+
+function getErrorMessage(errorCode: string): Node {
+    switch (errorCode) {
+        case 'tooManyVersions':
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/VersionError.js]: See above comment on `tooManyVersions`

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js (hunk 23 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/approval-comment-form/ApprovalCommentForm.js]: done. Using any.

File: src/components/ContentSidebar/ActivityFeed/comment/InlineDelete.js (hunk 24 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/comment/InlineDelete.js]: done

File: src/components/ContentSidebar/ActivityFeed/version/CollapsedVersion.js (hunk 25 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/CollapsedVersion.js]: done

File: src/components/ContentSidebar/ActivityFeed/version/Version.js (hunk 26 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/version/Version.js]: done

File: src/components/ContentSidebar/ActivityFeed/comment/InlineEdit.js (hunk 27 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/comment/InlineEdit.js]: done

File: src/components/ContentSidebar/ActivityFeed/activityFeedFlowTypes.js [hunk 40]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+export type InputState = {
+    approverSelectorContacts?: SelectorItems,
+    mentionSelectorContacts?: SelectorItems,
+    currentUser: User,
+    isDisabled?: boolean
+};
+
+export type Item = {
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activityFeedFlowTypes.js]: The only things it shares in common with `BoxItem` is
```
createdAt?: string,
createdBy: User,
id: string
```

Should I move over all of the props required into `BoxItem`?

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 41]
```diff
@@ -0,0 +1,110 @@
...
+                            )}
+                        </li>
+                    );
+                case 'file_version_error':
+                    // we currently only display this if errorCode is tooManyVersions
+                    if (item.errorCode !== 'tooManyVersions') {
+                        return null;
+                    }
+                    return (
+                        <li className='bcs-version-item' key={item.type + item.errorCode}>
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: No clue why. I didn't write this

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 42]
```diff
@@ -0,0 +1,188 @@
...
+    return feedState.reduce((collapsedFeedState, feedItem) => {
+        const previousFeedItem = collapsedFeedState.pop();
+
+        if (!previousFeedItem) {
+            return collapsedFeedState.concat([feedItem]);
+        }
+
+        if (
+            feedItem.type === 'file_version' &&
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 43]
```diff
@@ -0,0 +1,188 @@
...
+                }
+            ]);
+        }
+
+        return collapsedFeedState.concat([previousFeedItem, feedItem]);
+    }, []);
+}
+
+function shouldShowEmptyState(feedState: Array<Item>): boolean {
+    return feedState.length === 0 || (feedState.length === 1 && feedState[0].type === 'file_version');
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 44]
```diff
@@ -0,0 +1,188 @@
...
+                        }}
+                        isDisabled={inputState.isDisabled}
+                        approverSelectorContacts={approverSelectorContacts}
+                        mentionSelectorContacts={mentionSelectorContacts}
+                        className={`bcs-activity-feed-comment-input ${inputState.isDisabled ? 'bcs-is-disabled' : ''}`}
+                        createComment={this.createCommentHandler}
+                        createTask={handlers && handlers.tasks ? this.createTaskHandler : null}
+                        getApproverContactsWithQuery={
+                            handlers && handlers.contacts ? handlers.contacts.getApproverWithQuery : null
+                        }
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: done

File: src/components/ContentSidebar/ActivityFeed/activityFeedFlowTypes.js [hunk 45]
```diff
@@ -0,0 +1,53 @@
...
+ * @flow
+ * @file Flow types for Activity Feed
+ * @author Box
+ */
+
+import type { SelectorItems, User } from '../../../flowTypes';
+
+export type Comments = {
+    create: Function,
+    delete: Function
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activityFeedFlowTypes.js]: You're correct. Done!

File: src/components/ContentSidebar/ActivityFeed/icons/IconActivityFeedEmptyState.js [hunk 46]
```diff
@@ -0,0 +1,142 @@
+/* eslint-disable max-len */
+/**
+ * @flow
+ * @file IconActivityFeedEmptyState icon component
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/icons/IconActivityFeedEmptyState.js]: I'm going to leave as is, as this is specific to activity feed, not anything else inside of BRUI. If you can give me some tools for optimizing SVGs, I can optimize

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/AddApprovalFields.js [hunk 47]
```diff
@@ -0,0 +1,79 @@
...
+
+import type { SelectorItems } from '../../../../flowTypes';
+import messages from '../../../messages';
+
+type Props = {
+    approvalDate: ?number,
+    approvers: SelectorItems,
+    approverSelectorContacts: SelectorItems,
+    approverSelectorError: string,
+    formatMessage: Function,
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/approval-comment-form/AddApprovalFields.js]: done

File: src/components/ContentSidebar/ActivityFeed/approval-comment-form/AddApproval.js [hunk 48]
```diff
@@ -0,0 +1,64 @@
...
+    onApproverSelectorRemove: Function,
+    onApproverSelectorSelect: Function
+};
+
+const AddApproval = ({
+    approvalDate,
+    approvers,
+    approverSelectorContacts,
+    approverSelectorError,
+    formatMessage,
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/approval-comment-form/AddApproval.js]: done

**Comment by JustinHoldstock** [comment]
@phora, I've moved the location of the icon in BRUI. Once PR is accepted and new version published, we can upgrade our version of brui, reference the new icon, and remove the old one

File: src/components/ContentSidebar/ActivityFeed/utils/formatTaggedMessage.js (hunk 14 above)
Comment (priyajeet) [reply src/components/ContentSidebar/ActivityFeed/utils/formatTaggedMessage.js]: :trollface:
